elastic/kibana
# [Synthetics] implement project monitors read only

## Summary

Resolves https://github.com/elastic/kibana/issues/148966
Fixes https://github.com/elastic/kibana/issues/142807

<img width="1246" alt="Screen Shot 2023-01-30 at 11 03 53 AM" src="https://user-images.githubusercontent.com/11356435/215582541-866f9693-2b5b-402b-a5db-fbc4a1a36bc9.png">

Adds read-only view for Project monitors.

Allows only adjusting the `enabled` toggle.

Works with both browser and lightweight monitors.

Also improves the types for the monitor add/edit form, in order to avoid type coercion across form configuration. 

### Testing

1. Create at least 4 UI monitors, one of each type
2. Create at least 4 Project monitors, one of each type
3. Navigate to the edit page for each UI monitor. Ensure the experience remains the same, and you are not given a read-only experience.
4. Navigate to the edit page for each project monitor. Ensure all fields for all monitors, lightweight and browser, are all read-only. The only configuration that should be allowed is the `enabled` field.
5. Ensure the delete button remains enabled for both project and UI monitors
